### PR TITLE
First version of an options class

### PR DESCRIPTION
**Pull request by t-erxleben**
closing #13 
Should be enough for PoC and is extendable later on.

### CODE REVIEW

File: demo/backend/include/Color.hpp [hunk 1]
```diff
@@ -13,16 +13,16 @@ class Color
     Color();
 
     // construct by giving values for color channels
-    Color(uint8_t r, uint8_t g, uint8_t b);
+    Color(uint8_t const r, uint8_t const g, uint8_t const b);
```
Comment (JBenda) [inline demo/backend/include/Color.hpp:16]: Why `const`?
To enforce no modification in function ?
It's a copy anyway

File: demo/backend/include/Options.hpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        static Options* instance;
+        Options();
+
+
+        // private data members
+        Palette activePalette{};
+        Color bgColor{0xFFFFFF};
+
+    public:
+        // singleton should neither be copied nor assigned
```
Comment (JBenda) [inline demo/backend/include/Options.hpp]: then you should also delete the move assigment and move constructor
`Options(Options&&)` `Options&  operator=(Options&&)`

File: demo/backend/include/Palette.hpp [hunk 3]
```diff
@@ -13,9 +13,9 @@ class Palette
 public:
 
     // adds color to palette and returns its index
-    size_t add(Color col);
+    size_t const add(Color const col);
```
Comment (JBenda) [inline demo/backend/include/Palette.hpp]: This has literally no effect to anything!

**Review by JBenda — CHANGES_REQUESTED** [review]
The `const` thing is in at leas one point (see comment) useless, and confusing.
The singleton is still movable!

File: demo/backend/include/Options.hpp (hunk 2 above)
Comment (t-erxleben) [reply demo/backend/include/Options.hpp]: true; forgot about those

**Review by JBenda — APPROVED** [review]
good :)

**Pull request merged**